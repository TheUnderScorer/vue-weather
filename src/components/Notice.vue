<template>
	<div :class="type" class="notice">
		<section class="notice-text">
			<slot></slot>
		</section>
		<v-btn @click="$emit('notice-close')" class="notice-close" text x-small>
			x
		</v-btn>
	</div>
</template>

<script lang="ts">
    import { Component, Prop, Vue } from 'vue-property-decorator';
    import { NoticeType } from '@/components/types/NoticeType';

    @Component
    export default class Notice extends Vue
    {
        @Prop()
        public type!: NoticeType;
    }
</script>

<style lang="scss" scoped>
	@import '../scss/variables';

	.notice {
		padding: 0.5em;
		background-color: transparent;
		margin-bottom: 0.5em;
		display: flex;

		.notice-text {
			width: 90%;
		}

		&.error {
			background-color: $red;

			&, .notice-close {
				color: $white;
			}
		}
	}
</style>
