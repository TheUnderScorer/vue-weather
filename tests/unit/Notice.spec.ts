import { mount } from '@vue/test-utils';
import Notice from '@/components/Notice.vue';
import '@/plugins/vuetify';

describe( 'Notice.vue', () =>
{
    it( 'Renders without crashing', () =>
    {
        const component = mount( Notice, {
            slots: {
                default: 'Test notice!',
            },
        } );

        expect( component.html() ).toMatchSnapshot();
    } );

} );
