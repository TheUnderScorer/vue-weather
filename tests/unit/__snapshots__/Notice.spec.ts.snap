// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notice.vue Renders without crashing 1`] = `
<div class="notice">
  <section class="notice-text">Test notice!</section> <button type="button" class="notice-close v-btn v-btn--flat v-btn--text theme--light v-size--x-small"><span class="v-btn__content">
		x
	</span></button>
</div>
`;
